src/language-mode.coffee

```diff
@@ -298,15 +319,30 @@ class LanguageMode
     indentLevel = @suggestedIndentForBufferRow(bufferRow, options)
     @editor.setIndentationForBufferRow(bufferRow, indentLevel, options)
 
+
+  # Count the number of occurence of regex in string, given its tokenized
+  # version, and ignoring any occurence that start inside a quoted string
+  countRegex: (regex, string, tokenized) ->
+    count = 0
+    m = regex.searchSync(string)
+    while (m)
+      quoted = false
+      if scopes = tokenized.tokenAtBufferColumn(m[0].start)?.scopes
+        quoted = scope for scope in scopes when scope.match '^string.quoted'
```

Comment:
It would be good to break this loop once we find a match, and also to avoid performing a `RegExp` match here. What do you think about this:

```coffee
isQuoted = scopes.some (scope) -> scope.startsWith('string.quoted')
```